pkg/auth/authenticator/password/ldappassword/ldap.go

```diff
@@ -164,10 +126,10 @@ func (a *Authenticator) getIdentity(username, password string) (authapi.UserIden
 
 	// Build list of attributes to retrieve
 	attrs := util.NewStringSet(a.options.URL.QueryAttribute)
-	attrs.Insert(a.options.AttributeEmail...)
-	attrs.Insert(a.options.AttributeName...)
-	attrs.Insert(a.options.AttributePreferredUsername...)
-	attrs.Insert(a.options.AttributeID...)
+	attrs.Insert(a.options.IdentityAttributeMapping.EmailAttribute...)
```

Comment:
> I wasn't sure if Config type objects were OK to put methods on - seems like most that we have do not have methods but that's definitely something I want to do. It's unclear what refactor you're envisioning here, please elaborate.

Scheme'd objects do not have methods on them, but other valueobjects are allowed to have methods.